packages/core/src/utils/bigint.ts

```diff
@@ -76,3 +77,22 @@ export const bigIntClamp = (
 export const bigIntPercent = (base: BigNumberish, percent: bigint) => {
   return (BigInt(base) * percent) / 100n;
 };
+
+/**
+ * Useful if you want to use a string, such as a user's email address, as salt to generate a unique SmartAccount per user.
+ *
+ * example:
+ * ```
+ * const salt = stringToIndex("alice@example.com");
+ *
+ * export const account = new SimpleSmartContractAccount({
+ *   index: salt,
+ *   // other args omitted...
+ * });
+ * ```
+ *
+ * @param phrase -- any string value.
+ * @returns the bigint value of the hashed string
+ */
+export const stringToIndex = (phrase: string): bigint =>
+  BigInt(hashMessage(phrase));
```

Comment:
nit: we could simplify this a bit more to just be `keccak256(toHex(phrase))`

`hashMessage` adds the `0xEthereum Signed Message (len)` prefix to the bytes representation of the `phrase`. Not a big deal tbh, but reads a bit more clearly 🤷‍♂️

Reply:
Yeah, good call. I'll make that fix now